lib/cocoapods/downloader/cache.rb

```diff
@@ -115,12 +117,40 @@ def self.write_lock(location, block = Proc.new)
       #
       # @return [Void]
       #
-      def self.lock(location, lock_type, block = Proc.new)
-        lockfile = "#{location}.lock"
-        File.open(lockfile, File::CREAT, 0o644) do |f|
+      def self.lock(destination, lock_type, block = Proc.new)
+        lockfile = "#{destination}.lock"
+        f = nil
+        loop do
+          f.close if f
+          f = File.open(lockfile, File::CREAT, 0o644)
           f.flock(lock_type)
-          block.call location
+          break if Cache.validate_lock(f, lockfile)
         end
+        block.call destination
```

Comment:
should we wrap this in `begin /ensure`?

Reply:
Sounds sensible.

Reply:
Done, and strangely enough, Inch likes my documentation now. It seems a bit temperamental to me. 🤷‍♂️